tests/rptest/tests/end_to_end.py

```diff
@@ -172,16 +172,25 @@ def has_finished_consuming():
                    err_msg="Consumer failed to consume up to offsets %s after waiting %ds, last consumed offsets: %s." %\
                    (str(last_acked_offsets), timeout_sec, list(self.last_consumed_offsets)))
 
+    def await_num_produced(self, min_records, timeout_sec=30):
+        wait_until(lambda: self.producer.num_acked > min_records,
```

Comment:
nit: There are several tests in our tree that use this exact condition. For example, `mirror_maker_test.py.wait_for_n_messages`. If you feel like it, you could replace those instances with your `await_num_produced` method in a later PR :)

Reply:
Good thought. I'll keep that in mind and get to it if I don't forget! :)